### DISCUSSION TITLE
Why does `INCLUDED` record loaded files?

### DISCUSSION THREAD

**Discussion by SirWumpus** [ForthHub/discussion#181] [Q&A] [answered]
Forth 200x draft 19-1 section 11.6.2.2144.50 `REQUIRED`
> If the file specified by c-addr u has been INCLUDED or REQUIRED already ...

`REQUIRED` implies that the onus of recording loaded files is on `INCLUDED`, when there is no such mention in section 11.6.1.1718 `INCLUDED`.   Should it not be `REQUIRED`responsibility to record included files?  A clear separation of tasks.

Essentially `INCLUDED` loads untracked files, while `REQURIED` tracks and includes files.

**Comment by ruv** [comment, marked as answer]
> `REQUIRED` implies that the onus of recording loaded files is on `INCLUDED`, when there is no such mention in section 11.6.1.1718 INCLUDED. 

Yes. The idea was probably that "11.6.2.2144.50 REQUIRED" updates 11.6.1.1718 and the behavior of `INCLUDED`. In other cases this idea is expressed more explicitly.

> Should it not be `REQUIRED` responsibility to record included files? A clear separation of tasks.

I agree. I have never used Forth-2012 `REQUIRED` and `REQUIRE`. Far before Forth-2012 I used `REQUIRED-WORD ( sd.filename sd.wordname -- )` that loads (includes) the file if the given word is unavailable in the search order. Actually, it was named `REQUIRED` and `REQUIRE ( "wordname" "filename" -- ) `, because there were no such standard words back then.
